ui/components/UserPreferences.js

```diff
@@ -154,102 +151,108 @@ function ThemeToggler({ theme, themeSetter, notify, classes }) {
+
+    setCatalogContent(!catalogContent);
+    handleCatalogPreference(!catalogContent);
+  };
 
-class UserPreference extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      anonymousStats : props.anonymousStats,
-      perfResultStats : props.perfResultStats,
-      tabVal : 0,
-      userPrefs : ExtensionPointSchemaValidator("user_prefs")(),
-      providerType : '',
-      catalogContent : true,
-      extensionPreferences : {},
-      capabilitiesLoaded : false
-    };
-  }
-
-  handleCatalogContentToggle = () => {
-    this.props.toggleCatalogContent({ catalogVisibility : !this.state.catalogContent });
-    this.setState((state) => ({ catalogContent : !state.catalogContent }), () => this.handleCatalogPreference());
-  }
-
-  handleCatalogPreference = () => {
-    let body = Object.assign({}, this.state.extensionPreferences)
-    body["catalogContent"] = this.state.catalogContent
-    dataFetch("/api/user/prefs",
-      { credentials : "include",
-        method : "POST",
-        body : JSON.stringify({ usersExtensionPreferences : body })
-      },
+  const handleCatalogPreference = (catalogContent) => {
+    let body = Object.assign({}, extensionPreferences);
+    body["catalogContent"] = catalogContent;
+    dataFetch(
+      "/api/user/prefs",
+      { credentials : "include", method : "POST", body : JSON.stringify({ usersExtensionPreferences : body }) },
       () => {
-        const notify = this.props.notify;
-        notify({ message : `Catalog Content was ${this.state.catalogContent ? "enab" : "disab"}led`, event_type : EVENT_TYPES.SUCCESS })
+        const notify = props.notify;
+        notify({
+          message : `Catalog Content was ${catalogContent ? "enab" : "disab"}led`,
+          event_type : EVENT_TYPES.SUCCESS,
+        });
       },
-      this.handleError("There was an error sending your preference")
-    )
-  }
+      handleError("There was an error sending your preference")
+    );
+  };
 
-  handleToggle = (name) => () => {
-    const self = this;
-    if (name === 'anonymousUsageStats') {
-      self.setState((state) => ({ anonymousStats : !state.anonymousStats }), () => this.handleChange(name));
+  const handleToggle = (name) => () => {
+    if (name === "anonymousUsageStats") {
+      setAnonymousStats(!anonymousStats);
+      handleChange(name, !anonymousStats);
     } else {
-      self.setState((state) => ({ perfResultStats : !state.perfResultStats }), () => this.handleChange(name));
+      setPerfResultStats(!perfResultStats);
+      handleChange(name, !perfResultStats);
     }
-  }
+  };
 
-  handleError = (msg) => () => {
-    this.props.updateProgress({ showProgress : false });
-    const notify = this.props.notify;
-    notify({ message : msg, event_type : EVENT_TYPES.ERROR })
-  }
+  const handleError = (name) => () => {
+    props.updateProgress({ showProgress : false });
+    const notify = props.notify;
```

Comment:
In all over the codebase we can use `useNotification` hook with a functional component. And remove the `withNotify` wrapper as well.

Reply:
Sure will update that 👌